nats.go

```diff
@@ -1096,7 +1106,27 @@ func (nc *Conn) addURLToPool(sURL string, implicit bool) error {
 		}
 		sURL += defaultPortString
 	}
-	s := &srv{url: u, isImplicit: implicit}
+	var tlsName string
+	if implicit {
+		// Check to see if we do not have a url.User but current connected
+		// url does. If so copy over.
+		if u.User == nil && nc.url.User != nil {
+			u.User = nc.url.User
+		}
+		// We are checking to see if we have a secure connection and are
+		// adding an implicit server that just has an IP. If so we will remember
+		// the current hostname we are connected to.
+		if nc.Opts.Secure {
+			if addr := net.ParseIP(u.Hostname()); addr != nil {
+				if caddr := net.ParseIP(nc.url.Hostname()); caddr == nil {
```

Comment:
Not big of a deal, but addURLToPool is called with implicit == true from the processing of INFO with an array of urls. So it could be that we do this parseIP check many times for the same nc.url. Wonder if this could be moved to processInfo instead?

Reply:
You mean the one about the current connected correct? We could move it but we need the tlsName for when we create the *srv struct.

Reply:
processInfo() is the only one to call addURLToPool with implicit == true. processInfo runs under connection lock (when processing async INFO). The current nc.url cannot change while this processing is taking place, so I was suggesting that this `net.ParseIP(nc.url.Hostname()` be done once before the loop that calls addURLToPool(). But again, no big deal.

Reply:
Well I may look to hoist *srv instead of just url to nc.